### PR TITLE
Added staking page feature

### LINKED ISSUE #265: [Frontend] Add staking page for desktop
🚨 **Attention: Please read the guidelines before applying; otherwise, you will be ignored.** 🚫📋
0. Please, join the telegram group to keep in touch https://t.me/+m0IGRiBvLtsyMzIy 
1. Read guideline [here](https://gist.github.com/djeck1432/f722f5fe518c80b248ff609627da25e5)
2. Follow figma add staking page 
Screenshot 2024-11-25 at 17 54 09
3. Also update headers with adding `vault` page
Screenshot 2024-11-25 at 17 54 36
4. Make sure that page is responsive till mobile size [mobile design will be done in another task]

### PR DESCRIPTION
**Pull request by chigozzdevv**
Closes #265
image

I also imported the layout to the withdrawal page

image

### CODE REVIEW

**Comment by chigozzdevv** [comment]
@djeck1432 please review

**Comment by whateverfw** [comment]
@chigozzdevv fix failed CI PLS

File: frontend/src/pages/vault/VaultLayout.jsx
Comment (whateverfw) [inline frontend/src/pages/vault/VaultLayout.jsx:1]: please do bg color of this side-layout as in figma, also you can see different paddings also differ

File: frontend/src/pages/vault/stake/Stake.jsx [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+                  d="M6 9L12 15L18 9"
+                  stroke="currentColor"
+                  strokeWidth="2"
+                  strokeLinecap="round"
+                  strokeLinejoin="round"
+                />
+              </svg>
+            </div>
+
+            <div className="network-dropdown">
```
Comment (whateverfw) [inline frontend/src/pages/vault/stake/Stake.jsx:67]: image
can you please make it with rounded corners on option hover? currently it is squared

File: frontend/src/pages/vault/stake/Stake.jsx [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+              <CogIcon />
+            </div>
+            <div className="gas-fee-info">
+              <span className="gas-text">Gas fee:</span>
+              <span className="gas-amount">0.00 STRK</span>
+            </div>
+          </div>
+        </div>
+
+        <button className="stake-button">Stake</button>
```
Comment (whateverfw) [inline frontend/src/pages/vault/stake/Stake.jsx:95]: Please use our generic `Button` component from `components/ui`

File: frontend/src/pages/vault/stake/Stake.jsx [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+      <div className="staking-container">
+        <div className="balance-row">
+          <div className="balance-card">
+            <div className="balance-card-header">
+              <img src={STRK} alt="STRK" className="currency-icon" />
+              <span className="balance-label">STRK Balance</span>
+            </div>
+            <span className="balance-amount">0.046731</span>
+          </div>
+          <div className="balance-card">
```
Comment (whateverfw) [inline frontend/src/pages/vault/stake/Stake.jsx:27]: please make sure these cards are done according to figma, currently there are some mismatches in aligning with icons and font sizes

File: frontend/src/pages/vault/VaultLayout.jsx
Comment (whateverfw) [inline frontend/src/pages/vault/VaultLayout.jsx:1]: make sure to highligh active page just like in Figma

File: frontend/src/pages/vault/stake/Stake.jsx [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+          <div className="amount-input">
+            <input type="text" defaultValue="0.00" className="amount-field" />
+            <span className="currency">STRK</span>
+          </div>
+
+          <div className="apy-rate">$0.00 APY / year</div>
+          <hr />
+          <div className="gas-fee">
+            <div className="gas-icon">
```
Comment (whateverfw) [inline frontend/src/pages/vault/stake/Stake.jsx:85]: this icon differs from figma, can you make a shared component with withdraw and use the same icon used there?

File: frontend/src/pages/vault/stake/Stake.jsx [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+              <CogIcon />
+            </div>
+            <div className="gas-fee-info">
+              <span className="gas-text">Gas fee:</span>
+              <span className="gas-amount">0.00 STRK</span>
+            </div>
```
Comment (whateverfw) [inline frontend/src/pages/vault/stake/Stake.jsx:91]: you can create a shared component and use it here and on the withdraw page as it is the same

File: frontend/src/pages/vault/stake/Stake.jsx [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+                <div key={network.name} className="network-option" onClick={() => handleChange(network)}>
+                  <img src={network.image} alt={network.name} className="network-icon" />
+                  <span>{network.name}</span>
+                </div>
+              ))}
+            </div>
+          </div>
+
+          <div className="amount-input">
+            <input type="text" defaultValue="0.00" className="amount-field" />
```
Comment (whateverfw) [inline frontend/src/pages/vault/stake/Stake.jsx:78]: image
such thing can happen if I start writing a lot of numbers, part of the layout will be cut and everything will be shifted

File: frontend/src/App.jsx [hunk 7]
```diff
@@ -16,7 +16,8 @@ import Withdraw from 'pages/vault/withdraw/Withdraw';
 import { useWalletStore } from 'stores/useWalletStore'; 
 import { Notifier } from 'components/Notifier/Notifier';
 import { useConnectWallet } from 'hooks/useConnectWallet';
-
+import { Notifier } from 'components/Notifier/Notifier';
```
Comment (whateverfw) [inline frontend/src/App.jsx:19]: this is already imported above

**Review by whateverfw — CHANGES_REQUESTED** [review]

File: frontend/src/pages/vault/stake/Stake.jsx [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+  const networks = [{ name: 'Starknet', image: STRK }];
+
+  const handleChange = (e) => {
+    setSelectedNetwork(e.target.value);
+  };
+  return (
+    <VaultLayout>
+      <div className="staking-container">
+        <div className="balance-row">
+          <div className="balance-card">
```
Comment (whateverfw) [inline frontend/src/pages/vault/stake/Stake.jsx:20]: please use different name, not `balance-card` it conflicts with some items on other pages, and we'll deal with it later. for now just name it differently

**Review by whateverfw — CHANGES_REQUESTED** [review]

File: frontend/src/pages/vault/stake/Stake.jsx (hunk 3 above)
Comment (whateverfw) [reply frontend/src/pages/vault/stake/Stake.jsx:38]: still undersolved

File: frontend/src/pages/vault/VaultLayout.jsx
Comment (whateverfw) [reply frontend/src/pages/vault/VaultLayout.jsx:1]: still unresolved

File: frontend/src/pages/vault/stake/Stake.jsx (hunk 4 above)
Comment (whateverfw) [reply frontend/src/pages/vault/stake/Stake.jsx:105]: still unresolved

File: frontend/src/pages/vault/stake/Stake.jsx (hunk 8 above)
Comment (whateverfw) [reply frontend/src/pages/vault/stake/Stake.jsx:31]: still unresolved

File: frontend/src/components/StakeCard/GasFee.jsx
Comment (whateverfw) [inline frontend/src/components/StakeCard/GasFee.jsx:1]: this file should be in own folder, not inside StakeCard, also it should have its own css file with proper names, and not the ones from withdraw.css

File: frontend/src/pages/vault/stake/Stake.jsx
Comment (whateverfw) [inline frontend/src/pages/vault/stake/Stake.jsx:1]: please make top margins according to figma, currently gap is smaller

File: frontend/src/pages/vault/stake/stake.css
Comment (whateverfw) [inline frontend/src/pages/vault/stake/stake.css:1]: remove commented code, also use css variables (from global.css for colors)

**Review by whateverfw — CHANGES_REQUESTED** [review]

**Comment by djeck1432** [comment]
@chigozzdevv any updates?

**Comment by chigozzdevv** [comment]
> @chigozzdevv any updates?

Resolved sir

**Comment by chigozzdevv** [comment]
I'm still working to resolve the latest issues. So sorry for the delay @djeck1432 🙏. It'll be up soon.

**Comment by whateverfw** [comment]
@chigozzdevv so, did you resolve everything and I can check or it is still WIP?

**Comment by chigozzdevv** [comment]
> did you resolve everything and I can

Please check I resolved everything you asked of

**Review by whateverfw — APPROVED** [review]
@chigozzdevv @djeck1432 LGTM

**Pull request merged**